### PR TITLE
OSDF namespaces JSON: exclude caches that are down or inactive

### PR DESCRIPTION
**Pull request by matyasselmeci**
SOFTWARE-5440 and SOFTWARE-5471

### CODE REVIEW

File: src/stashcache.py [hunk 1]
```diff
@@ -543,20 +543,38 @@ def _namespace_dict(ns: Namespace):
...
+    def _resource_has_downed_cache(r: Resource, t: Topology):
+        if r.name not in t.present_downtimes_by_resource:
+            return False
+        downtimes = t.present_downtimes_by_resource[r.name]
+        for dt in downtimes:
+            try:
+                if XROOTD_CACHE_SERVER in dt.service_names:
+                    return True
+            except (KeyError, AttributeError):
+                continue
```
Comment (brianhlin) [inline src/stashcache.py:560]: Too bad we don't have a filter for this :(

**Review by brianhlin — APPROVED** [review]
One question otherwise LGTM

File: src/stashcache.py (hunk 1 above)
Comment (matyasselmeci) [reply src/stashcache.py:560]: A lot of these queries would be soooo much easier if I could do SQL queries on Topology...

File: src/stashcache.py [hunk 2]
```diff
@@ -543,20 +543,26 @@ def _namespace_dict(ns: Namespace):
         }
 
         for cache_name, cache_resource_obj in cache_resource_objs.items():
-            if resource_allows_namespace(cache_resource_obj, ns) and namespace_allows_cache_resource(ns, cache_resource_obj):
+            if resource_allows_namespace(cache_resource_obj, ns) and namespace_allows_cache_resource(ns,
+                                                                                                     cache_resource_obj):
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply src/stashcache.py]: Nope. Fixed.

**Pull request merged**